### PR TITLE
boards: stm32_min_dev enable I2C_2

### PR DESCRIPTION
**Pull request by pavlohamov**
Add support of i2c peripheral for mini_dev boards

Signed-off-by: Pavlo Hamov <pavlo_hamov@jabil.com>

### CODE REVIEW

**Comment by pavlohamov** [comment]
@ydamigos Could you be so kind to take a look?

**Review by ydamigos — APPROVED** [review]

**Review by erwango — CHANGES_REQUESTED** [review]
Please update board doc as well.

**Comment by pavlohamov** [comment]
@erwango Done

File: boards/arm/stm32_min_dev/doc/index.rst [hunk 1]
```diff
@@ -108,6 +108,16 @@ The stm32_min_dev board configuration supports the following hardware features:
 
 Other hardware features are not supported by the Zephyr kernel.
 
+I2C
```
Comment (erwango) [inline boards/arm/stm32_min_dev/doc/index.rst]: Oh! Tbh, I wasn't aware other pin configuration were not documented, I requested this update as I usually  do for other boards. So things are not quite homogeneous now on this board doc, but nvm, let's keep it now. Thanks

**Review by erwango — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Pull request merged**